pupil_src/shared_modules/video_capture/utils.py

```diff
@@ -233,12 +244,16 @@ def lookup_loc(self) -> str:
         return os.path.join(self.rec, f"{self.name}_lookup.npy")
 
     def fetch_videos(self) -> Iterator[Video]:
-        yield from (
+        videos = (
             Video(loc)
             for ext in self.video_exts
             for loc in glob.iglob(
                 os.path.join(self.rec, f"{self.name}*.{ext}"))
         )
+        # If not self.fill_gaps, we skip the broken videos
+        if not self.fill_gaps:
+            return [v for v in videos if v.is_valid]
+        return videos
```

Comment:
Previously, `fetch_videos()` always returned a generator. Now it only does if `fill_gaps==True`. Therefore, I suggest to `yield from` in both cases. Additionally, I would convert `[v for v in videos if v.is_valid]` from a list comprehension to a generator as well.